hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerMetadataInspector.java

```diff
@@ -392,6 +412,36 @@ private boolean checkAndRepair(JsonObject parent,
       errors.add(usedBytesError);
     }
 
+    // check and repair if db delete count mismatches delete transaction count.
+    final JsonElement pendingDeleteCountDB = dBMetadata.get(
+        OzoneConsts.PENDING_DELETE_BLOCK_COUNT);
+    final long dbDeleteCount = jsonToLong(pendingDeleteCountDB);
+    final JsonElement pendingDeleteCountAggregate
+        = aggregates.get(PendingDelete.COUNT);
+    final long deleteTransactionCount = jsonToLong(pendingDeleteCountAggregate);
+    if (dbDeleteCount != deleteTransactionCount) {
+      passed = false;
+
+      final BooleanSupplier deleteCountRepairAction = () -> {
+        final String key = containerData.getPendingDeleteBlockCountKey();
+        try {
+          // reset delete block count to 0 in metadata table
+          metadataTable.put(key, 0L);
```

Comment:
This should reset to the aggregate value, not 0. Looks like the tests did not catch this because `checkDBBlockAndByteCounts` needs to be modified to check the pending delete block count in the DB after repair.

Reply:
Good catch!  Just have pushed a commit to fix both problems.